pymodbus/client/base.py

```diff
@@ -122,7 +122,9 @@ def __init__(
 
         # Common variables.
         self.framer = self.params.framer(ClientDecoder(), self)
-        self.transaction = DictTransactionManager(self, **kwargs)
+        self.transaction = DictTransactionManager(
+            self, **{**kwargs, **{"retries": retries, "retry_on_empty": retry_on_empty}}
```

Comment:
```suggestion
            self, retries=retries, retry_on_empty=retry_on_empty, **kwargs
```

Reply:
I did not know you can do that. Looks better than my attempt to merge the dicts.
I am usually writing PHP and Java, so my Python code isn't super clean 😉